File: java/tests/StrataInspectorTest/src/tests/strata1/inspector/diagnostic/MockDiagnosticCheck.java
// ##########################################################################
// # File Name:		MockDiagnosticCheck.java
// # Copyright(C):	2005, Capital Group Companies, Inc. 
// #                All Rights Reserved.
// ##########################################################################

package strata1.inspector.diagnostic;


/**
 * 
 * @author 		
 *     ASG 
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public 
class MockDiagnosticCheck 
	extends DiagnosticCheck
{
	private boolean itsCheckFlag;
	private boolean itsRecoveryFlag;
	private boolean itsCanRecoverFlag;
	private boolean itsUnknownFlag;
	
	/************************************************************************
	 * Creates a new MockDiagnosticCheck. 
	 *
	 * @param name
	 */
	public MockDiagnosticCheck(
		String 	name,
		boolean check,
		boolean recovery,
		boolean canRecover,
		boolean unknown)
	{
		super( name );
		itsCheckFlag      = check;
		itsRecoveryFlag   = recovery;
		itsCanRecoverFlag = canRecover;
		itsUnknownFlag    = unknown;
	}
	
	/************************************************************************
	 * Returns the success message associated with the diagnostic check. 
	 *
	 * @return	success message
	 */
	public String getCheckSuccessMessage()
	{
		return getName() + ": Check has succeeded.";
	}

	/************************************************************************
	 * Returns the failure message associated with the diagnostic check. 
	 *
	 * @return	failure message
	 */
	public String getCheckFailureMessage()
	{
		return getName() + ": Check has failed.";
	}

	/************************************************************************
	 * Returns the success message associated with the diagnostic recovery. 
	 *
	 * @return	success message
	 */
	public String getRecoverySuccessMessage()
	{
		return getName() + ": Recovery has succeeded.";
	}

	/************************************************************************
	 * Returns the failure message associated with the diagnostic recovery. 
	 *
	 * @return	failure message
	 */
	public String getRecoveryFailureMessage()
	{
		return getName() + ": Recovery has failed.";
	}

	/************************************************************************
	 * {@inheritDoc} 
	 * @see com.capgroup.af.solar.application.DiagnosticCheck#runCheck()
	 */
	protected String runCheck() throws DiagnosticException
	{
		if ( !itsCheckFlag )
		{
			if ( !itsUnknownFlag )
				throw new DiagnosticException( getCheckFailureMessage() );
			else
				throw new RuntimeException( "Check: Unknown Exception" );
		}
		
		return getCheckSuccessMessage();
	}

	/************************************************************************
	 * {@inheritDoc} 
	 * @see com.capgroup.af.solar.application.DiagnosticCheck#runRecovery()
	 */
	protected String runRecovery() throws DiagnosticException
	{
		if ( !itsRecoveryFlag )
		{
			if ( !itsUnknownFlag )
				throw new DiagnosticException( getRecoveryFailureMessage() );
			else
				throw new RuntimeException( "Recovery: Unknown Exception" );
		}	
			
		return getRecoverySuccessMessage();
	}

	/************************************************************************
	 * {@inheritDoc} 
	 * @see com.capgroup.af.solar.application.DiagnosticCheck#isRecoverable()
	 */
	protected boolean isRecoverable()
	{
		return itsCanRecoverFlag;
	}

}


// ##########################################################################

File: legacy/csharp/tests/Strata.DiagnosticTest/Core/MockDiagnosticCheck.cs
//  ##########################################################################
//  # File Name: MockDiagnosticCheck.cs
//  # Copyright: 2011-2018, Sapientia Systems, LLC.
//  ##########################################################################

using System;

namespace Strata.Diagnostic.Core
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// $comments$
    /// </summary>
    /// <author>JFL</author>
    /// <conventions>$conventionspath$</conventions>
    ///  
    public 
    class MockDiagnosticCheck:
        DiagnosticCheck
    {
        public bool CheckFlag { get; set; }
        public bool RecoveryFlag { get; set; }
        public bool CanRecoverFlag { get; set; }
        public bool UnknownFlag { get; set; }


        //////////////////////////////////////////////////////////////////////
        /// <summary>
        /// Creates a new <c>DirectoryExistsCheck</c> instance.
        /// </summary>
        /// 
        public 
        MockDiagnosticCheck(
            String name,
            bool   check,
            bool   recover,
            bool   canRecover,
            bool   unknown):
            base( name )
        {
            CheckFlag      = check;
            RecoveryFlag   = recover;
            CanRecoverFlag = canRecover;
            UnknownFlag    = unknown;
        }

        //////////////////////////////////////////////////////////////////////
        /// <summary>
        /// Determines if the specified directory path exists.
        /// </summary>
        /// 
        protected override String 
        RunCheck()
        {
		    if ( !CheckFlag )
		    {
			    if ( !UnknownFlag )
				    throw new DiagnosticException( GetCheckFailureMessage() );
			    else
				    throw new Exception( "Check: Unknown Exception" );
		    }
		
		    return GetCheckSuccessMessage();
        }

        //////////////////////////////////////////////////////////////////////
        /// <summary>
        /// This diagnostic check does not perform recovery.
        /// </summary>
        /// 
        protected override String 
        RunRecovery()
        {
		    if ( !RecoveryFlag )
		    {
			    if ( !UnknownFlag )
				    throw new DiagnosticException( GetRecoveryFailureMessage() );
			    else
				    throw new Exception( "Recovery: Unknown Exception" );
		    }	
			
		    return GetRecoverySuccessMessage();
        }

        //////////////////////////////////////////////////////////////////////
        /// <summary>
        /// This diagnostic check does not perform recovery.
        /// </summary>
        /// 
        protected override bool 
        IsRecoverable()
        {
            return CanRecoverFlag;
        }

	    /************************************************************************
	     * Returns the success message associated with the diagnostic check. 
	     *
	     * @return	success message
	     */
	    public String 
        GetCheckSuccessMessage()
	    {
		    return Name + ": Check has succeeded.";
	    }

	    /************************************************************************
	     * Returns the failure message associated with the diagnostic check. 
	     *
	     * @return	failure message
	     */
	    public String 
        GetCheckFailureMessage()
	    {
		    return Name + ": Check has failed.";
	    }

	    /************************************************************************
	     * Returns the success message associated with the diagnostic recovery. 
	     *
	     * @return	success message
	     */
	    public String 
        GetRecoverySuccessMessage()
	    {
		    return Name + ": Recovery has succeeded.";
	    }

	    /************************************************************************
	     * Returns the failure message associated with the diagnostic recovery. 
	     *
	     * @return	failure message
	     */
	    public String 
        GetRecoveryFailureMessage()
	    {
		    return Name + ": Recovery has failed.";
	    }

    }
}

//  ##########################################################################
